AK/MemoryStream.cpp

```diff
@@ -270,13 +270,17 @@ ErrorOr<Bytes> AllocatingMemoryStream::next_write_range()
 
 void AllocatingMemoryStream::cleanup_unused_chunks()
 {
-    // FIXME: Move these all at once.
-    while (m_read_offset >= CHUNK_SIZE) {
+    auto const chunks_to_remove = m_read_offset / CHUNK_SIZE;
+    auto const chunks_to_move = m_chunks.size() - chunks_to_remove;
+
+    if (chunks_to_remove > 0) {
         VERIFY(m_write_offset >= m_read_offset);
 
-        auto buffer = m_chunks.take_first();
-        m_read_offset -= CHUNK_SIZE;
-        m_write_offset -= CHUNK_SIZE;
+        for (u64 i = 0; i < chunks_to_move; ++i)
+            m_chunks[i] = move(m_chunks[i + chunks_to_remove]);
```

Comment:
Doesn't this leave uninitialized(?) chunks at the very end of the Vector? Or do those get handled properly?

Reply:
> Doesn't this leave uninitialized(?) chunks at the very end of the Vector?
It absolutely does, I really wonder how it can work